libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -112,10 +112,9 @@ void DataFlowAnalyzer::operator()(FunctionDefinition& _fun)
 
 void DataFlowAnalyzer::operator()(ForLoop& _for)
 {
-	// Special scope handling of the pre block.
 	pushScope(false);
-	for (auto& statement: _for.pre.statements)
-		visit(statement);
+	// empty pre block in loops
```

Comment:
Please add a comment: `If the pre block was not empty, we would have to deal with more complicated scoping rules.`